anchor references seen as duplicate key

I'm not sure if this is a feature or a bug! :-)

When running yamllint on a yaml file with multiple anchors referenced, you get a duplicate key error. 

EG:

Given the following file:
```yaml
---
anchor_one: &anchor_one
  sequence_one:
    - sequence_one
    - sequence_two

anchor_two: &anchor_two
  sequence_two:
    - sequence_three
    - sequence_four

main:
  <<: *anchor_one
  <<: *anchor_two
  unicorn_settings:
    - non-sequence-sequence
```

Upon running yamllint you get the following error:
```bash
test.yml
  14:3      error    duplication of key "<<" in mapping  (key-duplicates)
```

Python thinks it's "valid" and has the desired data structure:

```python
>>> import yaml, pprint
>>> pp = pprint.PrettyPrinter(indent=2)
>>> file = open("test.yml", "r")
>>> yaml_dump = yaml.load(file)
>>> pp.pprint(yaml_dump)
{ 'anchor_one': { 'sequence_one': ['sequence_one', 'sequence_two']},
  'anchor_two': { 'sequence_two': ['sequence_three', 'sequence_four']},
  'main': { 'sequence_one': ['sequence_one', 'sequence_two'],
            'sequence_two': ['sequence_three', 'sequence_four'],
            'unicorn_settings': ['non-sequence-sequence']}}
```

[As near as I can tell, it seems to be supported by the YAML spec:](http://www.yaml.org/spec/1.2/spec.html#id2765878):

*When composing a representation graph from serialized events, an alias node refers to the most recent node in the serialization having the specified anchor. Therefore, anchors need not be unique within a serialization.*

--- a/tests/rules/test_key_duplicates.py
+++ b/tests/rules/test_key_duplicates.py
@@ -78,6 +78,15 @@ class KeyDuplicatesTestCase(RuleTestCase):
                    '    duplicates with\n'
                    '    many styles\n'
                    ': 1\n', conf)
+        self.check('---\n'
+                   'Merge Keys are OK:\n'
+                   'anchor_one: &anchor_one\n'
+                   '  one: one\n'
+                   'anchor_two: &anchor_two\n'
+                   '  two: two\n'
+                   'anchor_reference:\n'
+                   '  <<: *anchor_one\n'
+                   '  <<: *anchor_two\n', conf)
 
     def test_enabled(self):
         conf = 'key-duplicates: enable'
@@ -147,6 +156,15 @@ class KeyDuplicatesTestCase(RuleTestCase):
                    ': 1\n', conf,
                    problem1=(3, 1), problem2=(4, 1), problem3=(5, 3),
                    problem4=(7, 3))
+        self.check('---\n'
+                   'Merge Keys are OK:\n'
+                   'anchor_one: &anchor_one\n'
+                   '  one: one\n'
+                   'anchor_two: &anchor_two\n'
+                   '  two: two\n'
+                   'anchor_reference:\n'
+                   '  <<: *anchor_one\n'
+                   '  <<: *anchor_two\n', conf)
 
     def test_key_tokens_in_flow_sequences(self):
         conf = 'key-duplicates: enable'

--- a/yamllint/rules/key_duplicates.py
+++ b/yamllint/rules/key_duplicates.py
@@ -91,7 +91,9 @@ def check(conf, token, prev, next, nextnext, context):
         # This check is done because KeyTokens can be found inside flow
         # sequences... strange, but allowed.
         if len(context['stack']) > 0 and context['stack'][-1].type == MAP:
-            if next.value in context['stack'][-1].keys:
+            if (next.value in context['stack'][-1].keys and
+                    # `<<` is "merge key", see http://yaml.org/type/merge.html
+                    next.value != '<<'):
                 yield LintProblem(
                     next.start_mark.line + 1, next.start_mark.column + 1,
                     'duplication of key "%s" in mapping' % next.value)

